### ISSUE TITLE
In case of HealthStatus.Unhealthy exception is swallowed before writting to log

### ISSUE THREAD

**Issue by dziedrius** [dotnet/aspnetcore#20948]
Problem: API kind of suggests to pass exception in case you're returning Unhealthy result and then swallows it when writing to logs.

If you would look at HealthCheckResult, it has factory method Unhealthy, which accepts exception parameter:
```
public static HealthCheckResult Unhealthy(
           string description = null, 
           Exception exception = null, 
           IReadOnlyDictionary<string, object> data = null)
```

But if you will look at the the log message, that is produced for unhealthy status:
```
 private static readonly Action<ILogger, string, double, HealthStatus, string, Exception> _healthCheckEndUnhealthy = 
       LoggerMessage.Define<string, double, HealthStatus, string>(
                LogLevel.Error,
                EventIds.HealthCheckEnd,
                HealthCheckEndText);

private static readonly string HealthCheckEndText =
 "Health check {HealthCheckName} completed after {ElapsedMilliseconds}ms with status {HealthStatus} and '{HealthCheckDescription}'";
```

So I'm passing exception in hope, that it will be logged (as unhealthy response is logged for me) and it is not logged.

I could kind of log it myself for custom health checks, but there's plenty of already implemented healthchecks, that rely on exception being logged for them.

For example one of the popular project with bunch of health checks for various systems does it:

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.CosmosDb/CosmosDbHealthCheck.cs

I also saw similar issue, which has a bit different scope, but has exception part in common:
https://github.com/dotnet/aspnetcore/issues/9042

**Comment by mjolka** [comment]
I believe this is the same issue as #21588.

**Comment by pranavkm** [comment]
Duplicate of #21588 which is resolved as of 5.0-rc1.
